update docstring examples of `torch.testing.assert_close`

pmeier opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61031 remove UsageError in favor of ValueError
* #60639 change language from array to tensor
* #60638 relax type equality constraint
* #60637 Don't check stride by default
* #60636 Use maximum of tolerances in case of mismatching dtypes
* #60536 upcast to most precise dtype within their category before the comparison
* #60254 tests for diagnostics in callable `msg` in `torch.testing.assert_close`
* **#60163 update docstring examples of `torch.testing.assert_close`**
* #60091 Improve error messages of `torch.testing.assert_close` in case of mismatching values

Changes to the default error message in case of mismatching values need to be reflected in the examples given in the docstring. Normally this should be enforced by a [`doctest`](https://docs.python.org/3/library/doctest.html). @mruberry do you know why we don't have such a check?

Differential Revision: [D29556353](https://our.internmc.facebook.com/intern/diff/D29556353)

pmeier committed:
update docstring examples of `torch.testing.assert_close`

Changes to the default error message in case of mismatching values need to be reflected in the examples given in the docstring. Normally this should be enforced by a [`doctest`](https://docs.python.org/3/library/doctest.html). @mruberry do you know why we don't have such a check?

[ghstack-poisoned]

pmeier committed:
Update on "update docstring examples of `torch.testing.assert_close`"


Changes to the default error message in case of mismatching values need to be reflected in the examples given in the docstring. Normally this should be enforced by a [`doctest`](https://docs.python.org/3/library/doctest.html). @mruberry do you know why we don't have such a check?

[ghstack-poisoned]

pmeier committed:
Update on "update docstring examples of `torch.testing.assert_close`"


Changes to the default error message in case of mismatching values need to be reflected in the examples given in the docstring. Normally this should be enforced by a [`doctest`](https://docs.python.org/3/library/doctest.html). @mruberry do you know why we don't have such a check?

[ghstack-poisoned]

pmeier committed:
Update on "update docstring examples of `torch.testing.assert_close`"


Changes to the default error message in case of mismatching values need to be reflected in the examples given in the docstring. Normally this should be enforced by a [`doctest`](https://docs.python.org/3/library/doctest.html). @mruberry do you know why we don't have such a check?

[ghstack-poisoned]

pmeier committed:
Update on "update docstring examples of `torch.testing.assert_close`"


Changes to the default error message in case of mismatching values need to be reflected in the examples given in the docstring. Normally this should be enforced by a [`doctest`](https://docs.python.org/3/library/doctest.html). @mruberry do you know why we don't have such a check?

[ghstack-poisoned]

pmeier committed:
Update on "update docstring examples of `torch.testing.assert_close`"


Changes to the default error message in case of mismatching values need to be reflected in the examples given in the docstring. Normally this should be enforced by a [`doctest`](https://docs.python.org/3/library/doctest.html). @mruberry do you know why we don't have such a check?

[ghstack-poisoned]

mruberry reviewed (approved):
Neat.

We do have some doc testing, but I'm never sure how it works. @peterbell10 just started running some examples as tests, maybe check with him?

pmeier committed:
Update on "update docstring examples of `torch.testing.assert_close`"


Changes to the default error message in case of mismatching values need to be reflected in the examples given in the docstring. Normally this should be enforced by a [`doctest`](https://docs.python.org/3/library/doctest.html). @mruberry do you know why we don't have such a check?

[ghstack-poisoned]

pmeier committed:
Update on "update docstring examples of `torch.testing.assert_close`"


Changes to the default error message in case of mismatching values need to be reflected in the examples given in the docstring. Normally this should be enforced by a [`doctest`](https://docs.python.org/3/library/doctest.html). @mruberry do you know why we don't have such a check?

[ghstack-poisoned]

pmeier commented:
I've added a doctest for `torch.testing.assert_close`.

pmeier committed:
Update on "update docstring examples of `torch.testing.assert_close`"


Changes to the default error message in case of mismatching values need to be reflected in the examples given in the docstring. Normally this should be enforced by a [`doctest`](https://docs.python.org/3/library/doctest.html). @mruberry do you know why we don't have such a check?

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29556353).
